fusillade/clouddirectory.py

```diff
@@ -999,91 +991,151 @@ def _add_links_batch(self, links: List[str], link_type: str):
             )
         return operations
 
-    def _add_typed_links_batch(self, links: List[str], link_type: str):
+    def _add_typed_links_batch(self, links: List[str], object_type, link_type: str, attributes: Dict, incoming=False):
         """
         Attaches links to this object in CloudDirectory.
         """
         if not links:
             return []
-        parent_path = self.cd.get_obj_type_path(link_type)
+        link_path = self.cd.get_obj_type_path(object_type)
         batch_attach_typed_link = self.cd.batch_attach_typed_link
+        if incoming:
+            source, target = None, self.object_ref
+        else:
+            source, target = self.object_ref, None
         operations = []
         for link in links:
-            parent_ref = f"{parent_path}{self.hash_name(link)}"
-            attributes = {
-                'parent_type': link_type,
-                'child_type': self.object_type,
-            }
-            operations.append(
-                batch_attach_typed_link(
-                    parent_ref,
-                    self.object_ref,
-                    'association',
-                    attributes
-                )
-            )
+            if source:
+                target = f"{link_path}{self.hash_name(link)}"
+            else:
+                source = f"{link_path}{self.hash_name(link)}"
```

Comment:
This function isn't super clear to me, but it seems like `target = f"{link_path}{self.hash_name(link)}"` will always be set for at least the second iteration and all subsequent iterations.  Is that the intended behavior?

Reply:
You're right, good catch. Clearly there is a weakness in my tests that should have caught that. Fixed